cmd/test.go

```diff
@@ -0,0 +1,252 @@
+	Quiet      bool
+	TestCases  []string
+	Timeout    time.Duration
+}
+
+func newTestCmd(cmds ...*cobra.Command) *cobra.Command {
+	root := &cobra.Command{
+		Use:   "test",
+		Short: "Test subcommands provide test suite to evaluate current cluster setup",
+		Long:  `Test subcommands provide test suite to evaluate current cluster setup. Currently there is support for peer connection tests, beacon node and validator API.`,
+	}
+
+	root.AddCommand(cmds...)
+
+	return root
+}
+
+func bindTestFlags(cmd *cobra.Command, config *testConfig) {
+	cmd.Flags().StringVar(&config.OutputToml, "output-toml", "", "File path to which output can be written in TOML format.")
+	cmd.Flags().StringSliceVar(&config.TestCases, "test-cases", nil, "List of comma separated names of tests to be exeucted.")
+	cmd.Flags().DurationVar(&config.Timeout, "timeout", 24*time.Hour, "Execution timeout for all tests.")
+	cmd.Flags().BoolVar(&config.Quiet, "quiet", false, "Do not print test results to stdout.")
+}
+
+func mustOutputToFileOnQuiet(cmd *cobra.Command) error {
+	if cmd.Flag("quiet").Changed && !cmd.Flag("output-toml").Changed {
+		return errors.New("on --quiet, an --output-toml is required")
+	}
+
+	return nil
+}
+
+type testVerdict string
+
+const (
+	// boolean tests
+	testVerdictOk testVerdict = "OK"
+
+	// measurement tests
+	testVerdictGood testVerdict = "Good"
+	testVerdictAvg  testVerdict = "Avg"
+	testVerdictBad  testVerdict = "Bad"
+
+	// failed tests
+	testVerdictFail testVerdict = "Fail"
+)
+
+type categoryScore string
+
+const (
+	categoryScoreA categoryScore = "A"
+	categoryScoreB categoryScore = "B"
+	categoryScoreC categoryScore = "C"
+)
+
+type testResult struct {
+	Verdict     testVerdict `json:"verdict"`
+	Measurement string      `json:"measurement"`
+	Suggestion  string      `json:"suggestion"`
+	Error       string      `json:"error,omitempty"`
+}
+
+type testCaseName struct {
+	name  string
+	order uint
+}
+
+type testCategoryResult struct {
+	CategoryName  string                `json:"category_name"`
+	TestsExecuted map[string]testResult `json:"tests_executed"`
+	ExecutionTime Duration              `json:"execution_time"`
+	Score         categoryScore         `json:"score"`
+}
+
+func writeResultToFile(res testCategoryResult, path string) error {
+	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0o444)
+	if err != nil {
+		return errors.Wrap(err, "create/open file")
+	}
+	defer f.Close()
+	err = toml.NewEncoder(f).Encode(res)
+	if err != nil {
+		return errors.Wrap(err, "encode testCategoryResult to TOML")
+	}
+
+	return nil
+}
+
+func writeResultToWriter(res testCategoryResult, w io.Writer) error {
+	var lines []string
+
+	switch res.CategoryName {
+	case "peers":
+		lines = append(lines, "  ____                                                      ")
+		lines = append(lines, " |  __ \\                                                    ")
+		lines = append(lines, " | |__) |___   ___  _ __  ___                               ")
```

Comment:
Can we have the ASCII art as a multiline const instead?

Reply:
I can put them as a function that returns []string? If I use it as a single string with new lines it's a bit harder to attach the score next to it later.

Reply:
Yeah that's better